src/async_impl/body.rs

```diff
@@ -0,0 +1,98 @@
+use futures::{Stream, Poll, Async};
+use bytes::Bytes;
+
+pub struct Body {
```

Comment:
It seems like a shame to lose some of the `From` conversions on the synchronous body when dealing with the asynchronous one directly.